content/intro-to-storybook/vue/es/composite-component.md

```diff
@@ -0,0 +1,276 @@
+
+![default and pinned tasks](/intro-to-storybook/tasklist-states-1.png)
+
+Dado que los datos de `Task` pueden enviarse asincrónicamente, **también** necesitamos un estado de cargando para renderizar en ausencia de alguna conexión. Además, también se requiere un estado vacío para cuando no hay tareas.
+
+![empty and loading tasks](/intro-to-storybook/tasklist-states-2.png)
+
+## Empezar la configuración
+
+Un componente compuesto no es muy diferente de los componentes básicos que contiene. Crea un componente `TaskList` y un archivo de historia que lo acompañe: `src/components/TaskList.vue` y `src/components/TaskList.stories.js`.
+
+Comienza con una implementación aproximada de la `TaskList`. Necesitarás importar el componente `Task` del capítulo anterior y pasarle los atributos y acciones como entrada.
+
+```html
+
+<!--src/components/TaskList.vue-->
+<template>
+  <div>
+    <div class="list-items" v-if="loading">loading</div>
+    <div class="list-items" v-if="noTasks && !this.loading">empty</div>
+    <div class="list-items" v-if="showTasks">
+      <task
+        v-for="(task, index) in tasks"
+        :key="index"
+        :task="task"
+        @archiveTask="$emit('archiveTask', $event)"
+        @pinTask="$emit('pinTask', $event)"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+  import Task from './Task';
+  export default {
+    name: 'task-list',
+    props: {
+      loading: {
+        type: Boolean,
+        default: false,
+      },
+      tasks: {
+        type: Array,
+        default: () => []
+      },
+    },
+    components: {
+      Task,
+    },
+    computed: {
+      noTasks() {
+        return this.tasks.length === 0;
+      },
+      showTasks() {
+        return !this.loading && !this.noTasks;
+      },
+    },
+  };
+</script>
+```
+
+A continuación, crea los estados de prueba de `Tasklist` en el archivo de historia.
+
+```javascript
+//src/components/TaskList.stories.js
+import TaskList from "./TaskList";
+import { taskData, actionsData } from "./Task.stories";
+
+const paddedList = () => {
+  return {
+    template: '<div style="padding: 3rem;"><story/></div>'
+  };
+};
+export default {
+  title: "TaskList",
+  excludeStories: /.*Data$/,
+  decorators: [paddedList]
+};
+
+export const defaultTasksData = [
+  { ...taskData, id: "1", title: "Task 1" },
+  { ...taskData, id: "2", title: "Task 2" },
+  { ...taskData, id: "3", title: "Task 3" },
+  { ...taskData, id: "4", title: "Task 4" },
+  { ...taskData, id: "5", title: "Task 5" },
+  { ...taskData, id: "6", title: "Task 6" }
+];
+export const withPinnedTasksData = [
+  ...defaultTasksData.slice(0, 5),
+  { id: "6", title: "Task 6 (pinned)", state: "TASK_PINNED" }
+];
+
+// default TaskList state
+export const Default = () => ({
+  components: { TaskList },
+  template: `<task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
+  props: {
+    tasks: {
+      default: () => (defaultTasksData)
```

Comment:
@IvanSotelo  we can do without the extra parenthesis in the snippet, to avoid inducing some confusion, reason being if the reader as already setup prettier, which is becoming widely adopted as code formatter will start having a "field day" (pardon the bad pun) adding unnecessary warnings. It should be 

```javascript
default: () => defaultTasksData
```

same logic goes below with the `WithPinnedTasks `